US-3382, US-3381 Added a test sample and perform coding standard changes

pfoster-youitv opened the pull request:
## Change list

Added some sample tests as BasicViewsTest.java
Adjusted code to meet Google's coding standards.
 
## Types of changes

- Added test sample class: BasicViewsTest.java
- Added a test utility class:  TestUtility.java
- Corrected Google code style violations (over 280 errors fixed, 10 warnings do remain around import groups)
- Modified sanityTest.java to use the SimpleViews app.
- Modified paths in the test classes to be relative to the SimpleViews app (once it is included in the package)
- Added remote/YouiEngineCapabilityType.java

## Details

The added test class can double as a sanity test of sorts and Google coding style was applied.

## To Do

- [ ] The sample test class uses a sample app that needs to be added as both an app and apk.
- [ ] Document that Android users will want to either supply their own AVD or point to a device instead.

simongranger commented:
Can you look at adding our caps here: https://github.com/YOU-i-Labs/appium-java-client/tree/develop/src/main/java/io/appium/java_client/remote?

pfoster-youitv commented:
The client doesn't seem to require it because the caps are just key-value pairs passed down to the server which is what uses them. But it is something we could look at adding (not in this PR), especially if we intend to grow the custom caps we allow.

pfoster-youitv commented:
Added in the caps interface for youi Engine.
Also updated our code to use the caps strings
Also updated the SanityTest class to use the SimpleViews app

Still need to adjust the paths to be relative and add in the app/apk files.

simongranger commented on src/main/java/io/appium/java_client/YouiEngine/YouiEngineDriver.java:
```diff
@@ -2,36 +2,36 @@
 
 import io.appium.java_client.AppiumDriver;
 import io.appium.java_client.YouiEngine.internal.JsonToYouiEngineElementConverter;
+import io.appium.java_client.remote.YouiEngineCapabilityType;
```
This should be in alphabetical order

pfoster-youitv replied:
Not a Google coding style violation.

simongranger commented on src/test/java/io/appium/java_client/YouiEngine/BasicViewsTest.java:
```diff
@@ -0,0 +1,307 @@
+            capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, "6.0");
+            capabilities.setCapability(YouiEngineCapabilityType.APP_PLATFORM, "android");
+        } else {
+            capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, "9.3");
```
Not sure this adds any value, it may actually make it more complicated if the simulator version gets bumped up.
Here is an example if I set it 9.1:

An unknown server-side error occurred while processing the command. Original error: Could not find a device to launch. You requested 'iPhone 6 (9.1) [', but the available devices were:,"iPhone 6 (9.3) [AA313554-873B-4E20-976F-367BFEA25319] (Simulator)", ...

pfoster-youitv replied:
Not necessary. The version should be specific to what is being tested. People using this will need to define their own caps.

simongranger commented on src/test/java/io/appium/java_client/YouiEngine/BasicViewsTest.java:
```diff
@@ -0,0 +1,307 @@
+        if (isAndroid) {
+            capabilities.setCapability(AndroidMobileCapabilityType.AVD, "nexus5intel");
+            capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "Android");
+            capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, "6.0");
```
See platform_version comment on line 88

simongranger commented on src/test/java/io/appium/java_client/YouiEngine/util/AppiumTest.java:
```diff
@@ -54,46 +67,43 @@ public static void init(YouiEngineDriver webDriver, URL driverServerAddress) {
-            capabilities.setCapability("youiAppAddress", "localhost");
-            capabilities.setCapability("youiAppPlatform", "iOS");
-            capabilities.setCapability("deviceName", "iPhone Simulator");
+            capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, "9.3");
```
Not sure this adds any value, it may actually make it more complicated if the simulator version gets bumped up.
Here is an example if I set it 9.1:

An unknown server-side error occurred while processing the command. Original error: Could not find a device to launch. You requested 'iPhone 6 (9.1) [', but the available devices were:,"iPhone 6 (9.3) [AA313554-873B-4E20-976F-367BFEA25319] (Simulator)", ...

simongranger commented on src/test/java/io/appium/java_client/YouiEngine/util/AppiumTest.java:
```diff
@@ -54,46 +67,43 @@ public static void init(YouiEngineDriver webDriver, URL driverServerAddress) {
+            capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "Android");
+            capabilities.setCapability(YouiEngineCapabilityType.APP_PLATFORM, "android");
+            capabilities.setCapability(AndroidMobileCapabilityType.AVD, "nexus5intel");
+            capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, "6.0.1");
```
See comment on line 87

simongranger commented on src/main/java/io/appium/java_client/YouiEngine/YouiEngineDriver.java:
```diff
@@ -48,26 +48,33 @@ public void swipe(int startx, int starty, int endx, int endy, int duration) {
             super.execute("lock", getCommandImmutableMap("secs", seconds));
         } else if (appPlatform.equals(ANDROID)) {
             this.lock();
+            this.lock();
```
lock twice?

simongranger commented:
Can you maybe add a description in each test file describing its purpose?

pfoster-youitv commented:
In the next PR which will focus on documentation and cleaning up the comments. This needs to go in to pass the Google code style rules. No more in this PR.

simongranger commented:
+1
